### ISSUE TITLE
JavaScript support

### ISSUE THREAD

**Issue by lpil** [gleam-lang/json#1]
- [ ] Use JS' built in JSON encoder/decoder.

**Comment by mrkutly** [comment]
@lpil I tried taking a crack at this (mostly just for fun as a way of learning more gleam) and have a draft PR here:

**Comment by lpil** [comment]
Wonderful. I couldn't see the failing test?

**Comment by mrkutly** [comment]
It might just be an environment issue on my end then. I was seeing these errors:

```
gleam_json_test.decode_bits_unexpected_format_test: Error:
        Error(UnexpectedFormat([DecodeError(expected: "Int", found: "Tuple of 0 elements", path: [])]))
        should equal
        Error(UnexpectedFormat([DecodeError(expected: "Int", found: "List", path: [])]))

gleam_json_test.decode_unexpected_format_test: Error:
        Error(UnexpectedFormat([DecodeError(expected: "Int", found: "Tuple of 0 elements", path: [])]))
        should equal
        Error(UnexpectedFormat([DecodeError(expected: "Int", found: "List", path: [])]))
```

When I went through those code paths, it seemed like something was going wrong in `gleam/dynamic`, but I am super new to gleam so I could be way off.

**Comment by pascalwengerter** [comment]
@lpil can this be closed? `v0.5.0` suggests so :)

**Comment by lpil** [comment]
Oh yes! Thank you
